ibc-derive/src/client_state/traits/client_state_common.rs

```diff
@@ -104,25 +123,56 @@ pub(crate) fn impl_ClientStateCommon(
                 }
             }
 
+            fn serialize_path(&self, path: impl Into<#Path>) -> core::result::Result<#PathBytes, #ClientError> {
+                match self {
+                    #(#serialize_path_impl),*
+                }
+            }
+
+            fn verify_membership_raw(
+                &self,
+                prefix: &#CommitmentPrefix,
+                proof: &#CommitmentProofBytes,
+                root: &#CommitmentRoot,
+                path: #PathBytes,
+                value: Vec<u8>,
+            ) -> core::result::Result<(), #ClientError> {
+                match self {
+                    #(#verify_membership_raw_impl),*
+                }
+            }
+
             fn verify_membership(
                 &self,
                 prefix: &#CommitmentPrefix,
                 proof: &#CommitmentProofBytes,
                 root: &#CommitmentRoot,
-                path: #Path,
+                path: impl Into<#Path>,
                 value: Vec<u8>,
             ) -> core::result::Result<(), #ClientError> {
                 match self {
                     #(#verify_membership_impl),*
                 }
             }
```

Comment:
we can remove `verify_membership` and `verify_non_membership` from trait implementations. They are default implementations.

Reply:
Good point. I included them in the macro to make it complete and functional, even if someone chooses not to use the default implementation (though we don't expect this to happen).